File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count=0, maxcount=0;

        for(int i=0; i < nums.size(); i++){
            if(nums[i] == 1){
                count++;
            }
            if(nums[i] == 0){
                if(maxcount <= count){
                    maxcount = count;
                    count=0;
                }
                else{
                    count =0;
                }
            }
            if(i == nums.size()-1 && count > maxcount){
                maxcount = count;
            }
        }

        return maxcount;
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        if(nums.length == 0) return 0;
        if(nums.length == 1) return nums[0];
        int i = 0, j = 0 , ans = 0;
       
        while(i < nums.length){
           if(nums[i] == 0){
            j = i + 1;
            
           }
           if(nums[i] == 1)ans = Math.max(ans, (i - j + 1));
           i++;
        }
        return ans;
    }
}